### ISSUE TITLE
File with mtime equal to 0 are ignored by Cling.

### ISSUE THREAD

**Issue by pcanal** [root-project/root#14372]
### Check duplicate issues.

- [X] Checked for duplicates

### Description

This problem was discovered during a build through Spack that was using the Spack build cache.  In this build cache Spack 
has elected to reset the modification time of ***all*** headers files to be 0 linux time.

It turns out in that case, Cling/Clang/LLVM refused to actually uses any of the system headers coming from the spack-provided compiler build.  So nothing is properly defined not even `__GLIBCXX__`

The visible consequence is:
```
... rootcling_stage1  ...  G__Core.cxx ... 
Warning in cling::IncrementalParser::CheckABICompatibility():
  Failed to extract C++ standard library version.
In file included from input_line_2:1:
/scratch/pcanal/spack-root-badter/tmp/spack-stage/spack-stage-root-develop-4gn6o5fpka65i75lxkr6h63al2u2hkmg/spack-build-4gn6o5f/etc/cling/Interpreter/RuntimeUniverse.h:175:47: error: use of undeclared identifier 'std'
      void copyArray(T* src, void* placement, std::size_t size) {
                                              ^
/scratch/pcanal/spack-root-badter/tmp/spack-stage/spack-stage-root-develop-4gn6o5fpka65i75lxkr6h63al2u2hkmg/spack-build-4gn6o5f/etc/cling/Interpreter/RuntimeUniverse.h:176:14: error: use of undeclared identifier 'std'
        for (std::size_t i = 0; i < size; ++i)
             ^
In file included from input_line_3:1:
In file included from /usr/include/assert.h:35:
In file included from /usr/include/features.h:514:
/usr/include/gnu/stubs.h:7:11: fatal error: 'gnu/stubs-32.h' file not found
# include <gnu/stubs-32.h>
          ^~~~~~~~~~~~~~~~
Error: Error loading the default rootcling header files.
ninja: build stopped: subcommand failed.
```

### Reproducer

```
cat /var/tmp/a.cxx 
void a()
{
   fprintf(stdout, "Running the script a.cxx\n");
}

touch -m a -d '1969-12-31 18:00:00.000000000' /var/tmp/a.cxx 

ls -l /var/tmp/a.cxx 
-rw-r--r-- 1 pcanal us_cms 63 Dec 31  1969 /var/tmp/a.cxx

root.exe -b -l -q /var/tmp/a.cxx 

Processing /var/tmp/a.cxx...
input_line_8:1:10: fatal error: file '/var/tmp/a.cxx' modified since it was first processed
#include "/var/tmp/a.cxx"
         ^
```

### ROOT version

  | Welcome to ROOT 6.31/01                        https://root.cern |
  | (c) 1995-2023, The ROOT Team; conception: R. Brun, F. Rademakers |
  | Built for linuxx8664gcc on Jan 12 2024, 21:36:24                 |
  | From heads/master@9645b3aa                                       |
  | With g++ (Spack GCC) 13.1.0                                      |

### Installation method

SPACK with build Cache (but technically any).

### Operating system

Linux but likely any

### Additional context

The problem was introduced during the upgrade to LLVM 16.

**Comment by hahnjo** [comment]
If this problem was introduced with the upgrade to LLVM 16, then it cannot affect 6.30 (with LLVM 13) and consequently there is nothing to fix for 6.30.04.

**Comment by pcanal** [comment]
Thanks for fixing this clicko (typo with a click) error :)
